optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/Joiners.java

```diff
@@ -147,10 +151,99 @@
+     *         A's and B's intervals (as defined by the function mapping) overlap
+     */
+    public static <A, B, Property_ extends Comparable<Property_>> BiJoiner<A, B> overlaps(
+            Function<A, Property_> leftStartMapping,
+            Function<A, Property_> leftEndMapping,
+            Function<B, Property_> rightStartMapping,
+            Function<B, Property_> rightEndMapping) {
+        return AbstractBiJoiner.merge(Joiners.lessThan(leftStartMapping, rightEndMapping),
+                Joiners.greaterThan(leftEndMapping, rightStartMapping));
+    }
+
+    /**
+     * Syntactic sugar for {@link #overlaps(Function, Function, Function, Function)} where
+     * both elements of the tuple (A,B) are the same type and have the same interval
+     * defining function. Identical to
+     * {@code overlaps(startMapping, endMapping, startMapping, endMapping)}.
+     * 
+     * @param <A> the type of both the first and second argument
+     * @param <Property_> the type used to define the interval, comparable
+     * 
+     * @param startMapping maps the argument to the start point of its interval (inclusive)
+     * @param endMapping maps the argument to the end point of its interval (exclusive)
+     * 
+     * @return see {@link #overlaps(Function, Function, Function, Function)}
+     */
+    public static <A, Property_ extends Comparable<Property_>> BiJoiner<A, A> overlaps(
+            Function<A, Property_> startMapping,
+            Function<A, Property_> endMapping) {
+        return overlaps(startMapping, endMapping, startMapping, endMapping);
+    }
+
+    /**
+     * For pairs of intervals [a,b) and [c,d), returns only those pairs where [a,b) contains [c,d) (or, in other
+     * words, for any x in [c, d), x is also in [a,b)).
+     * These are exactly the pairs where {@code a <= c} and {@code b >= d} (or in other words, when the first interval
+     * starts before the second one starts and the first interval ends after the second interval ends).
+     * 
+     * @param <A> the type of the first argument
+     * @param <B> the type of the second argument
+     * @param <Property_> the type used to define the interval, comparable
+     * 
+     * @param leftStartMapping maps the first argument to the first interval start point (inclusive)
+     * @param leftEndMapping maps the first argument to the first interval end point (exclusive)
+     * @param rightStartMapping maps the second argument to the second interval start point (inclusive)
+     * @param rightEndMapping maps the second argument to the second interval end point (exclusive)
+     * 
+     * @return never null, an indexed joiner that filters the constraint stream to only include elements (A,B) where
+     *         A's interval contains B's interval (as defined by the function mapping)
+     */
+    public static <A, B, Property_ extends Comparable<Property_>> BiJoiner<A, B> during(
```

Comment:
-1 on "during" as method name because it presumes the notion of time.
This method applies not just to LocalDate etc, but also Integer, Long, BigDecimal and anything else that is Comparable.
Not just intervals, but any type of range.

Reply:
I've opened a zulip chat on this, this is something we need to resolve before merging I think, as it gets written in stone in the API.